homeassistant/components/madvr/coordinator.py

```diff
@@ -0,0 +1,66 @@
+"""Coordinator for handling data fetching and updates."""
+
+import logging
+from typing import TYPE_CHECKING, Any
+
+from madvr.madvr import Madvr
+
+from homeassistant.const import CONF_HOST, CONF_MAC, CONF_PORT
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.device_registry import DeviceInfo
+from homeassistant.helpers.update_coordinator import DataUpdateCoordinator
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+if TYPE_CHECKING:
+    from . import MadVRConfigEntry
+
+
+class MadVRCoordinator(DataUpdateCoordinator[dict[str, Any]]):
+    """Madvr coordinator for Envy (push-based API)."""
+
+    if TYPE_CHECKING:
```

Comment:
We usually don't do the if type checking here, not sure what the impact is but if we can remove it, that would be nice